### PR TITLE
Enumerable test RSpec

### PR DESCRIPTION
**Pull request by Aduda-Boaz**
The PR is about testing the Enumerable methods with RSpec. The project is built by @fortuneonyeka and @Aduda-Boaz for the purposes of learning.

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    end
+    it 'when block given' do
+      result = arry.my_count { |x| x > 5 }
+      expect(result).to eql(2)
+    end
+  end
+
+  describe 'my_map' do
+    it 'when block is given' do
+      result = range.map { |x| x * x }
```
Comment (enaburekhan) [inline spec/enumerables_spec.rb]: Use the custom method - `my_map` for the test. Instead of the regular `map` enumerable method.

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    it 'when block given' do
+      result = (5..10).my_inject { |sum, n| sum + n }
+      expect(result).to eql(45)
+    end
+    it 'when block has a method and an argument' do
+      result = (1..3).my_inject(1) { |x, i| x * i }
+      expected = (1..3).my_inject(1) { |x, i| x * i }
+      expect(result).to eql(expected)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerables_spec.rb]: Kindly add tests that also covers a negative case scenario ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)).

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      expect(arry.my_count).to eql(5)
+    end
+    it 'return even numbers when block is given' do
+      expect(arry.my_count(&:even?)).to eql(2)
+    end
+    it 'when block given' do
+      result = arry.my_count { |x| x > 5 }
+      expect(result).to eql(2)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerables_spec.rb]: Kindly also add negative test case scenarios([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi team!
Great job so far:clap::clap:.**

**Major highlights**

- The readme is very descriptive and professional
- The codebase is clean, easy to read and understand

However, we still need to make a few changes to get the project approved:muscle:.

**To improve**

- [x] Check my inline comments on the PR below and make the required changes.

Kindly request another review when you are ready.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Review by enaburekhan — APPROVED** [review]
# Status: Approve:heavy_check_mark:
**Hi team!
Congratulations:clap::clap:. Thank you for implementing all the changes earlier requested.
Move on with the next project:walking:. And don't forget to merge the PR:blush:.**

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.
